server/src/main/java/org/opensearch/index/codec/composite/LuceneDocValuesConsumerFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.codec.composite;
+
+import org.apache.lucene.codecs.DocValuesConsumer;
+import org.apache.lucene.codecs.lucene90.Lucene90DocValuesConsumerWrapper;
+import org.apache.lucene.index.SegmentWriteState;
+
+import java.io.IOException;
+
+import static org.opensearch.index.codec.composite.composite99.Composite99Codec.COMPOSITE_INDEX_CODEC_NAME;
+
+/**
+ * A factory class that provides a factory method for creating {@link DocValuesConsumer} instances
+ * based on the specified composite codec.
+ *
+ * @opensearch.experimental
+ */
+public class LuceneDocValuesConsumerFactory {
+
+    public static DocValuesConsumer getDocValuesConsumerForCompositeCodec(
+        String compositeCodec,
+        SegmentWriteState state,
+        String dataCodec,
+        String dataExtension,
+        String metaCodec,
+        String metaExtension
+    ) throws IOException {
+
+        switch (compositeCodec) {
+            case COMPOSITE_INDEX_CODEC_NAME:
+                return new Lucene90DocValuesConsumerWrapper(state, dataCodec, dataExtension, metaCodec, metaExtension)
```

Comment:
This is always used in write path, correct? In that case, should it ever be called for any codec other than the latest one?

Reply:
yeah, agreed. How do you propose we get the latest codec? Do you think we can have a static variable maybe in CompositeCodecFactory?
Also, now that I think, should we keep this classes within composite? Maybe some day, some other codec wants to use the wrapper for consumer?

Reply:
Can we follow OpenSearch does - right now we manually set default codec as Lucene99 for example right

Reply:
done, please check now!